### PR TITLE
Contributing lib/ansible/modules/network/cloudengine/ce_snmp_community.py module to manage HUAWEI data center CloudEngine

### PR DESCRIPTION
**Pull request by QijunPan**
add ce_snmp_community

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

ce_snmp_community.py
##### ANSIBLE VERSION

```
ansible 2.3.0
config file = /etc/ansible/ansible.cfg
configured module search path = ['/home/ansible-2.1.1.0/lib/ansible/modules/core/network/cloudengine', '/home/ansible-2.1.1.0/lib/ansible/module_utils', '/home/ansible-2.1.1.0/lib/ansible/modules', '/home/ansible-2.1.1.0', '/home/ansible-2.1.1.0/lib/ansible/utils/']
```

##### SUMMARY

```
TASK [config snmp community] ***************************************************
task path: /usr1/code/OpenNESS/code/current/KVM/intg/test/testcases/test-ce_snmp_community.yml:49
Using module file /usr/local/lib/python2.7/site-packages/ansible-2.3.0-py2.7.egg/ansible/modules/core/network/cloudengine/ce_snmp_community.py
<ce_6850> ESTABLISH LOCAL CONNECTION FOR USER: root
<ce_6850> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1487053553.0-180342122436386 `" && echo ansible-tmp-1487053553.0-180342122436386="` echo $HOME/.ansible/tmp/ansible-tmp-1487053553.0-180342122436386 `" ) && sleep 0'
<ce_6850> PUT /tmp/tmpXAe92b TO /root/.ansible/tmp/ansible-tmp-1487053553.0-180342122436386/ce_snmp_community.py
<ce_6850> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1487053553.0-180342122436386/ /root/.ansible/tmp/ansible-tmp-1487053553.0-180342122436386/ce_snmp_community.py && sleep 0'
<ce_6850> EXEC /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1487053553.0-180342122436386/ce_snmp_community.py; rm -rf "/root/.ansible/tmp/ansible-tmp-1487053553.0-180342122436386/" > /dev/null 2>&1 && sleep 0'
changed: [ce_6850] => {
    "changed": true, 
    "end_state": {}, 
    "existing": {
        "snmp community": {
            "community_info": []
        }, 
        "snmp v3 group": {
            "community_info": [
                {
                    "accessRight": "write", 
                    "aclNumber": "2000", 
                    "communityName": "%^%#H8RZ1uZ*g+\\,3i:\"H*D1%&2F&+)jJBP`pfU_u)LHPpz7C#&m'WH.uj8UF*L~}R@SHc3,-TY#8])t|h3,%^%#"
                }
            ], 
            "group_info": []
        }
    }, 
    "invocation": {
        "module_args": {
            "access_right": "write", 
            "acl_number": "2000", 
            "auth_pass": null, 
            "authorize": false, 
            "community_mib_view": null, 
            "community_name": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "group_name": null, 
            "host": "ce_6850", 
            "notify_view": null, 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "port": 12345, 
            "provider": null, 
            "read_view": null, 
            "security_level": null, 
            "ssh_keyfile": null, 
            "state": "present", 
            "timeout": 10, 
            "transport": null, 
            "use_ssl": false, 
            "username": "rootDC", 
            "validate_certs": true, 
            "write_view": null
        }, 
        "module_name": "ce_snmp_community"
    }, 
    "proposed": {
        "access_right": "write", 
        "acl_number": "2000", 
        "community_name": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
        "state": "present"
    }, 
    "updates": [
        "snmp-agent community write ****** acl 2000"
    ]
}

```

### CODE REVIEW

File: lib/ansible/modules/network/cloudengine/ce_snmp_community.py [hunk 1]
```diff
@@ -0,0 +1,941 @@
...
+        security_level = module.params['security_level']
+        acl_number = module.params['acl_number']
+        read_view = module.params['read_view']
+        write_view = module.params['write_view']
+        notify_view = module.params['notify_view']
+
+        community_name = module.params['community_name']
+        access_right = module.params['access_right']
+
+        if group_name and not security_level:
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_snmp_community.py]: There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
required_together
```

File: lib/ansible/modules/network/cloudengine/ce_snmp_community.py [hunk 2]
```diff
@@ -0,0 +1,941 @@
...
+            module.fail_json(
+                msg='Error: Please input group_name and security_level at the same time.')
+        if not group_name and security_level:
```
Comment (gundalow) [inline lib/ansible/modules/network/cloudengine/ce_snmp_community.py]: There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
required_together
```

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/cloudengine/ce_snmp_community.py (hunk 2 above)
Comment (QijunPan) [reply lib/ansible/modules/network/cloudengine/ce_snmp_community.py]: These issues have been fixed, thank you very much.

**Comment by QijunPan** [comment]
@gundalow   These problems have been fixed， can  this PR merged?

**Review by gundalow — APPROVED** [review]

**Pull request merged**